Microsoft.Toolkit.Uwp.UI.Media/Geometry/CanvasPathGeometry.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Numerics;
+using System.Text;
+using Microsoft.Graphics.Canvas;
+using Microsoft.Graphics.Canvas.Brushes;
+using Microsoft.Graphics.Canvas.Geometry;
+using Microsoft.Toolkit.Uwp.UI.Media.Geometry.Common;
+using Microsoft.Toolkit.Uwp.UI.Media.Geometry.Parsers;
+using Windows.UI;
+
+namespace Microsoft.Toolkit.Uwp.UI.Media.Geometry
+{
+    /// <summary>
+    /// Helper Class for creating Win2d objects
+    /// </summary>
+    public static class CanvasPathGeometry
+    {
+        /// <summary>
+        /// Parses the Path data string and converts it to CanvasGeometry.
+        /// </summary>
+        /// <param name="resourceCreator">ICanvasResourceCreator</param>
+        /// <param name="pathData">Path data</param>
+        /// <param name="logger">(Optional) For logging purpose. To log the set of
```

Comment:
Is the logger only useful for the Sample app and helping to construct samples? Wondering if we want that as a separate method or just show those examples hard-coded within the sample app? @Sergio0694 what's the extra overhead if all the calls are null - does the app using this compiler just optimize all those out?

Reply:
The logger is mainly used to construct the samples.
Added separate method without the logger.

Reply:
@michael-hawker Those calls will be rewritten by the C# compiler like this:
```csharp
if (logger != null)
{
    logger.ApendLine(...);
}
```
So there's no overhead per se in the execution, though the codegen is still worse since you do end up with the extra code for those branches, even when they're not taken. Now, if the entire method was to be inlined in the callee, and the argument was `null` as a constant (as in, either a literal `null`, or just not specified and using the default), then the JIT would be able to remove those branches entirely. But since the method is not particularly small, it's not generally a great candidate for inlining. Of course, given the type of method, this would still not likely be a performance concern, since the actual construction of the geometry would be far slower than just the possible overhead of more cache misses for the instruction data caused by the worse codegen. We're talking like not even in the same ballpark, so that's fine.

I do have a question regarding the API design per se - if the logger method is only useful for the samples, then why is this being added as part of the public API area? Shouldn't that code only be specifically as a helper in the sample app, with the APIs in the `.Media` package having no logger parameter at all? 🤔

Reply:
@Sergio0694 you are right, I will make the logger methods as internal and only the methods without the logger will be exposed as APIs.
Will push the changes tonight.

Reply:
Awesome, thanks! I'm also thinking other than internal, can they actually be defined directly in the sample app? As in, even though they still wouldn't be exposed publicly, it'd be nice to make the binary size of the package smaller by not having those methods at all, if we really only use them in the sample app. Is that doable without too much work? 🤔

Reply:
@Sergio0694 Removed the logger altogether from CanvasPathGeometry and internal classes. Now using a ICanvasPathReceiver in the SampleApp class to read the path data from CanvasGeometry and using it to construct the CanvasPathBuilder commands.